js/trbinance.js

```diff
@@ -0,0 +1,1334 @@
+        const amount = this.safeString2 (trade, 'q', 'qty');
+        const cost = this.safeString2 (trade, 'quoteQty', 'baseQty');  // inverse futures
+        const marketId = this.safeString (trade, 'symbol');
+        const symbol = this.safeSymbol (marketId, market);
+        let id = this.safeString2 (trade, 't', 'a');
+        id = this.safeString2 (trade, 'id', 'tradeId', id);
+        let side = undefined;
+        const orderId = this.safeString (trade, 'orderId');
+        if ('m' in trade) {
+            // this is reversed intentionally
+            if (trade['m']) {
+                side = 'sell';
+            } else {
+                side = 'buy';
+            }
+        } else if ('isBuyerMaker' in trade) {
+            if (trade['isBuyerMaker']) {
+                side = 'sell';
+            } else {
+                side = 'buy';
+            }
+        } else if ('side' in trade) {
+            side = this.safeStringLower (trade, 'side');
+        } else {
+            if ('isBuyer' in trade) {
+                if (trade['isBuyer']) {
+                    side = 'buy';
+                } else {
+                    side = 'sell';
+                }
+            }
+        }
+        let fee = undefined;
+        if ('commission' in trade) {
+            fee = {
+                'cost': this.safeString (trade, 'commission'),
+                'currency': this.safeCurrencyCode (this.safeString (trade, 'commissionAsset')),
+            };
+        }
+        let takerOrMaker = undefined;
+        if ('m' in trade) {
+            if (trade['m']) {
+                takerOrMaker = 'maker';
+            } else {
+                takerOrMaker = 'taker';
+            }
+        }
+        if ('maker' in trade) {
+            if (trade['maker']) {
+                takerOrMaker = 'maker';
+            } else {
+                takerOrMaker = 'taker';
+            }
+        }
+        if ('isMaker' in trade) {
+            if (trade['isMaker']) {
+                takerOrMaker = 'maker';
+            } else {
+                takerOrMaker = 'taker';
+            }
+        }
+        return this.safeTrade ({
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'id': id,
+            'order': orderId,
+            'type': undefined,
+            'side': side,
+            'takerOrMaker': takerOrMaker,
+            'price': price,
+            'amount': amount,
+            'cost': cost,
+            'fee': fee,
+        }, market);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            // 'symbol': market['id'],
+            'symbol': market['base'] + market['quote'], // Symbol needs to be without underscore for the endpoint
```

Comment:
We should not pass unified currency codes to the exchange, instead we should use currency ids here, otherwise it violated the following guideline:  ​https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sending-market-ids

​So we should change it like this:

```JavaScript
'symbol': market['baseId'] + market['quoteId']
```